assets/src/scripts/components/dailyValueHydrograph/legend.js

```diff
@@ -0,0 +1,94 @@
+// functions to facilitate DV legend creation for a d3 plot
+import {createStructuredSelector} from 'reselect';
+
+import {getLayout} from './selectors/layout';
+import {getLegendMarkerRows} from './selectors/legend-data';
+
+import config from '../../config';
+import { mediaQuery } from '../../utils';
+import {link} from '../../lib/d3-redux';
+
+
+/**
+ * Create a simple legend
+ *
+ * @param {Object} div - d3 selector where legend should be created
+ * @param {Object} legendMarkerRows - Array of rows. Each row should be an array of legend markers.
+ * @param {Object} layout - width and height of svg.
+ */
+export const drawSimpleLegend = function(div, {legendMarkerRows, layout}) {
+    div.selectAll('.legend-svg').remove();
+
+    if (!legendMarkerRows.length || !layout) {
```

Comment:
We don't need to check layout as it will always return an object with properties. I know it's in the hydrograph legend but consider removing it.